satellite/console/service.go

```diff
@@ -894,11 +897,19 @@ func (s *Service) CreateProject(ctx context.Context, projectInfo ProjectInfo) (p
 		s.log.Debug(fmt.Sprintf("could not add promotional coupon for user %s", auth.User.ID.String()), zap.Error(Error.Wrap(err)))
 		return p, nil
 	}
-	if len(cards) == 0 {
+
+	balance, err := s.accounts.Balance(ctx, auth.User.ID)
```

Comment:
project seems to be already created at this moment

Reply:
Its about applying the proper coupon here, or what are you referring to?

Reply:
If we move fetch of customer balance and credit cards before creating a project, we will prevent project creation when you don't have a card or your balance is too low, or there is an error during retrieving this info so we won't add a coupon. IMHO we want to keep project and coupon in sync as much as we can, right? It's just a suggestion, I am not sure what is the actual requirement

Reply:
As discussed with @rikysya we are going to address this after this PR in an entire rewrite of it.